PvPManager/src/main/java/me/NoChance/PvPManager/Listeners/EntityListener.java

```diff
@@ -119,6 +123,11 @@ private void onDamageActions(final Player attacker, final Player defender) {
 			}
 		}
 		if (Settings.isInCombatEnabled()) {
+			if (wg != null && Settings.borderHoppingVulnerable()) {
+				if (!Settings.borderHoppingResetCombatTag() && (!wg.hasAllowPvPFlag(attacker) && (!wg.hasAllowPvPFlag(defender)))) {
```

Comment:
Did you test if this works? Seems to me like it wouldn't renew the tag in case there are no flags too, instead of only when the flag is set to deny.